modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/StemmerOverrideTokenFilterFactory.java

```diff
@@ -57,19 +57,23 @@ public TokenStream create(TokenStream tokenStream) {
 
     static void parseRules(List<String> rules, StemmerOverrideFilter.Builder builder, String mappingSep) {
```

Comment:
@cbuescher yes, the only reason I didn't write any tests was that I didn't find any unit test classes for this token filter. But I agree that it needs it, even more now that there is some extra logic.

I'll take a look how you test token filters and address your comments.

Reply:
I think there's no need to emulate what other token filters are testing at this point (unless you like to), just adding `StemmerOverrideTokenFilterFactoryTests` to `org.elasticsearch.analysis.common` and add tests for the static method should be okay. Looks like most similar "*FactoryTests" extend `ESTokenStreamTestCase`, which might be a good idea to do as well, then it might be easier to add more test functionality going forward. There's some more info on how to run certain tests only in the [TESTING.asciidoc](https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc#test-case-filtering), if you have other questions let me know.

Reply:
@cbuescher is it really worth to use the complicated `StemmerOverrideMap` lookup API to verify that `parseRules` added the right key and value?

I would rather simply try to instantiate `StemmerOverrideTokenFilterFactory` and check if it doesn't throw (I would do it with a loop and `expectThrows` as you don't seem to use JUnit's parameterized tests).

Happy case (valid rules) could be tested with analysis output of its `TokenStream`.

Reply:
Sure, use anything you think makes sense. The `StemmerOverrideMap` API was just a thought.  Any approach that is easy to set up and read and that doesn't need to spin up a whole cluster (i.e. integration test) is fine. Testing some valid and invalid cases is the most important thing I think.